ppb/engine.py

```diff
@@ -95,12 +95,18 @@ def start(self):
     def main_loop(self):
         while self.running:
             time.sleep(0)
-            now = time.monotonic()
-            self.signal(events.Idle(now - self._last_idle_time))
-            self._last_idle_time = now
-            while self.events:
-                self.publish()
-            self.manage_scene()
+            self.run_one_iteration()
+
+    def run_one_iteration(self):
```

Comment:
`run_one_iteration` isn't a very clear name; perhaps `run_one_frame`?

Reply:
That's possibly deceptive. One iteration of the event loop does not correspond to one animation frame or simulation frame anymore.

Reply:
Maybe `run_one_loop_iteration`? The name is important here, because this is the public API that will be exposed to non-native event loops.

Reply:
```
pathunstrom Today at 7:12 AM
I'll make a command decision, with the note that we're still unstable and can fix it later, let's call it loop_once and have it working.
```

Reply:
done